src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/privateApi/CaseloadControllerTest.kt

```diff
@@ -0,0 +1,87 @@
+package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.resource.privateApi
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.mockito.kotlin.reset
+import org.mockito.kotlin.times
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest
+import org.springframework.boot.test.mock.mockito.MockBean
+import org.springframework.http.MediaType
+import org.springframework.test.context.ActiveProfiles
+import org.springframework.test.context.ContextConfiguration
+import org.springframework.test.context.junit.jupiter.SpringExtension
+import org.springframework.test.context.web.WebAppConfiguration
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers.status
+import org.springframework.test.web.servlet.setup.MockMvcBuilders
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.config.ControllerAdvice
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.ApprovalCase
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.CaseloadService
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.caseload.ApproverCaseloadService
+
+@ExtendWith(SpringExtension::class)
+@ActiveProfiles("test")
+@WebMvcTest(controllers = [CaseloadController::class])
+@AutoConfigureMockMvc(addFilters = false)
+@ContextConfiguration(classes = [CaseloadController::class])
+@WebAppConfiguration
+class CaseloadControllerTest {
+
+  @MockBean
+  private lateinit var caseloadService: CaseloadService
+
+  @MockBean
+  private lateinit var approverCaseloadService: ApproverCaseloadService
+
+  @Autowired
+  private lateinit var mvc: MockMvc
+
+  @Autowired
+  private lateinit var mapper: ObjectMapper
+
+  @BeforeEach
+  fun reset() {
+    reset(caseloadService, approverCaseloadService)
+
+    mvc = MockMvcBuilders
+      .standaloneSetup(CaseloadController(caseloadService, approverCaseloadService))
+      .setControllerAdvice(ControllerAdvice())
+      .build()
+  }
+
+  @Test
+  fun `get licences for approval`() {
+    val request = listOf(
+      "MDI",
+      "ABC",
+    )
+
+    val approvalCase = ApprovalCase(
+      licenceId = 1L,
+    )
+
+    whenever(approverCaseloadService.getApprovalNeeded(request)).thenReturn(listOf(approvalCase))
+
+    val response = mvc.perform(
+      post("/caseload/get-approval-needed")
```

Comment:
**suggestion:** could group by user `/caseload/prison-approver/approval-needed` ?

Reply:
or: `/caseload/prison-approvals/to-review ?`

Reply:
I prefer the first suggestion as it is clear who the caseload is for and what the caseload is for.